### PR TITLE
8264139: Suppress removal warnings for Security Manager methods

### PR DESCRIPTION
**Pull request by kevinrushforth**
This PR adds the necessary `@SuppressWarnings("removal")` annotations for the recently-integrated security manager deprecation, [JEP 411](https://openjdk.java.net/jeps/411). See openjdk/jdk#4073.

There are four commits:

1. 678b026 : A patch generated by @wangweij to add annotations to the runtime (`modules/*/src/main/java`) using the same automated tooling that he used as part of the implementation of JEP 411.
2. 9698e87 : Same as above for the tests.
3. 1c42cf8 : Manually removes a pair of unused imports, one of which (a static import) was causing a removal warning.
4. 4f87d18 : Manually reduced the scope of the annotation where it was added to an entire class, or to a large method where only a small part of the method uses a deprecated method. This was done using similar techniques to the following fixes that Weijun did in openjdk/jdk#4172.
 
The first two commits represent the bulk of the changes. Other than scanning to ensure that there are no obvious errors, and testing, they probably don't need the same level of scrutiny as the manual changes do.

I tested this on all three platforms by doing a build / test with `JDK_HOME` set to a local JDK 17 ea build that includes the fix for JEP 411. I ran the build with `gradle -PLINT=removal` and verified that there were removal warnings for the security manager APIs without this fix and none with this fix.

NOTE: The following files under `modules/javafx.web/src/android` and `modules/javafx.web/src/ios` were not processed by the automated tool. As I have no way to compile them, I chose not to manually fix them either, but doing so would be trivial as a follow-up fix if desired.

```
modules/javafx.web/src/android/java/com/sun/webkit/Timer.java
modules/javafx.web/src/android/java/com/sun/webkit/WebPage.java
modules/javafx.web/src/android/java/javafx/scene/web/WebEngine.java
modules/javafx.web/src/ios/java/javafx/scene/web/ExportedJavaObject.java
modules/javafx.web/src/ios/java/javafx/scene/web/HTMLEditorSkin.java
modules/javafx.web/src/ios/java/javafx/scene/web/JS2JavaBridge.java
modules/javafx.web/src/ios/java/javafx/scene/web/WebEngine.java
```

/reviewers 2
/contributor add weijun
/contributor add kcr

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264139](https://bugs.openjdk.java.net/browse/JDK-8264139): Suppress removal warnings for Security Manager methods

### Reviewers
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - no project role) 🔄 Re-review required (review applies to 4f87d1873429f11f35c988cddafdf6d834366104)
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - **Reviewer**)
 * [Ambarish Rapte](https://openjdk.java.net/census#arapte) (@arapte - **Reviewer**)

### Contributors
 * Weijun Wang `<weijun@openjdk.org>`
 * Kevin Rushforth `<kcr@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/528/head:pull/528` \
`$ git checkout pull/528`

Update a local copy of the PR: \
`$ git checkout pull/528` \
`$ git pull https://git.openjdk.java.net/jfx pull/528/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 528`

View PR using the GUI difftool: \
`$ git pr show -t 528`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/528.diff">https://git.openjdk.java.net/jfx/pull/528.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.graphics/src/main/java/javafx/scene/Node.java [hunk 1]
```diff
@@ -9966,7 +9966,7 @@ Accessible getAccessible() {
         if (accessible == null) {
             accessible = Application.GetApplication().createAccessible();
             accessible.setEventHandler(new Accessible.EventHandler() {
-                @SuppressWarnings({"removal","deprecation"})
+                @SuppressWarnings("removal")
```
Comment (wangweij) [inline modules/javafx.graphics/src/main/java/javafx/scene/Node.java:9969]: Was this "deprecation" already useless before this change?

File: modules/javafx.web/src/main/java/com/sun/webkit/network/HTTP2Loader.java [hunk 2]
```diff
@@ -412,11 +411,13 @@ private HTTP2Loader(WebPage webPage,
         };
 
         // Run the HttpClient in the page's access control context
-        this.response = AccessController.doPrivileged((PrivilegedAction<CompletableFuture<Void>>) () -> {
+        @SuppressWarnings("removal")
+        CompletableFuture<Void> tmpResponse = AccessController.doPrivileged((PrivilegedAction<CompletableFuture<Void>>) () -> {
```
Comment (wangweij) [inline modules/javafx.web/src/main/java/com/sun/webkit/network/HTTP2Loader.java:415]: Is "var" enough?

File: modules/javafx.fxml/src/main/java/com/sun/javafx/fxml/ModuleHelper.java [hunk 3]
```diff
@@ -31,16 +31,17 @@
...
 
-@SuppressWarnings("removal")
 public class ModuleHelper {
     private static final Method getModuleMethod;
     private static final Method getResourceAsStreamMethod;
 
     private static final boolean verbose;
 
     static {
-        verbose = AccessController.doPrivileged((PrivilegedAction<Boolean>) () ->
```
Comment (wangweij) [inline modules/javafx.fxml/src/main/java/com/sun/javafx/fxml/ModuleHelper.java:41]: You can merge this assignment with the declaration on line 38. Or you can keep this so the check of verbose is in the same block with its assignment.

File: modules/javafx.graphics/src/main/java/com/sun/glass/ui/Screen.java [hunk 4]
```diff
@@ -41,7 +40,9 @@
     private static final int dpiOverride;
 
     static {
-        dpiOverride = AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Integer.getInteger("com.sun.javafx.screenDPI", 0)).intValue();
+        @SuppressWarnings("removal")
```
Comment (wangweij) [inline modules/javafx.graphics/src/main/java/com/sun/glass/ui/Screen.java:43]: Combine assignment and declaration?

File: modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/LinuxArch.java [hunk 5]
```diff
@@ -28,16 +28,17 @@
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 
-@SuppressWarnings("removal")
 public class LinuxArch {
 
     private static final int bits;
 
     static {
-        bits = AccessController.doPrivileged((PrivilegedAction<Integer>) () -> {
```
Comment (wangweij) [inline modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/LinuxArch.java:36]: Combine?

File: modules/javafx.web/src/main/java/com/sun/webkit/network/NetworkContext.java [hunk 6]
```diff
@@ -240,12 +242,14 @@ private static int fwkGetMaximumHTTPConnectionCountPerHost() {
         private static final Permission modifyThreadGroupPerm = new RuntimePermission("modifyThreadGroup");
         private static final Permission modifyThreadPerm = new RuntimePermission("modifyThread");
 
+        @SuppressWarnings("removal")
```
Comment (wangweij) [inline modules/javafx.web/src/main/java/com/sun/webkit/network/NetworkContext.java:245]: Maybe move this onto the 1st line inside the method?

**Review by wangweij — APPROVED** [review]
All change looks good. Some personal style preferences comments.

File: modules/javafx.graphics/src/main/java/javafx/scene/Node.java (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/javafx/scene/Node.java:9969]: Yes. The code to which it applied was removed back in JDK 9, but we forgot to remove the annotation.

File: modules/javafx.fxml/src/main/java/com/sun/javafx/fxml/ModuleHelper.java (hunk 3 above)
Comment (kevinrushforth) [reply modules/javafx.fxml/src/main/java/com/sun/javafx/fxml/ModuleHelper.java:41]: I think I'll keep this one as is to minimize changes.

File: modules/javafx.graphics/src/main/java/com/sun/glass/ui/Screen.java (hunk 4 above)
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/com/sun/glass/ui/Screen.java:43]: Good idea. This allows the static block to be removed.

File: modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/LinuxArch.java (hunk 5 above)
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/LinuxArch.java:36]: Good idea. This allows the static block to be removed.

File: modules/javafx.web/src/main/java/com/sun/webkit/network/HTTP2Loader.java (hunk 2 above)
Comment (kevinrushforth) [reply modules/javafx.web/src/main/java/com/sun/webkit/network/HTTP2Loader.java:415]: Yes, I'll change it.

**Comment by aghaisas** [comment]
When built with JDK 17ea26 + `gradle -PLINT=removal` - I cross verified on my macBook (macOS 10.15) that -
- without this fix - the warnings are generated 
- with this fix - the warnings are not generated

**Review by aghaisas — APPROVED** [review]

**Review by arapte — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**